plugins/coverimage.koplugin/main.lua

```diff
@@ -67,9 +71,39 @@ end
 
 function CoverImage:createCoverImage(doc_settings)
    if self.enabled and not doc_settings:readSetting("exclude_cover_image") == true then
-        local image = self.ui.document:getCoverPageImage()
-        if image then
+        local cover_image = self.ui.document:getCoverPageImage()
+        if cover_image then
+            local s_w, s_h = Device.screen:getWidth(), Device.screen:getHeight()
+            local i_w, i_h = cover_image:getWidth(), cover_image:getHeight()
+            local scale_factor = math.min(s_w / i_w, s_h / i_h)
+
+            if self.cover_image_background == "none" or scale_factor == 1 then
+                cover_image:writePNG(self.cover_image_path, false)
+                return
+            end
+
+            local scaled_w, scaled_h = i_w * scale_factor, i_h * scale_factor
+            local cover_image = RenderImage:scaleBlitBuffer(cover_image, scaled_w, scaled_h)
```

Comment:
Do those work properly? In https://github.com/koreader/koreader/pull/6709 using "float" in blitbuffer caused image to be "projected".

Reply:
For me it seems, that the images are clear, but a `math.floor` can not hurt.